src/tz_windows.rs

```diff
@@ -1,16 +1,74 @@
+use std::mem;
+
+use windows_bindings::Windows::Win32::System::{
+    SystemServices::{TIME_ZONE_ID_DAYLIGHT, TIME_ZONE_ID_STANDARD, TIME_ZONE_ID_UNKNOWN},
+    Time::{GetDynamicTimeZoneInformation, TIME_ZONE_ID_INVALID},
+};
+
 #[path = "windows_bindings.rs"]
 #[allow(missing_debug_implementations, clippy::undocumented_unsafe_blocks)]
 mod windows_bindings;
-use windows_bindings::Windows::Globalization::Calendar;
 
-impl From<windows_core::Error> for crate::GetTimezoneError {
-    fn from(orig: windows_core::Error) -> Self {
-        crate::GetTimezoneError::IoError(std::io::Error::new(std::io::ErrorKind::Other, orig))
+/// <https://learn.microsoft.com/en-us/windows/win32/api/timezoneapi/nf-timezoneapi-getdynamictimezoneinformation>
+/// <https://learn.microsoft.com/en-us/windows/win32/api/timezoneapi/ns-timezoneapi-dynamic_time_zone_information>
+pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
+    let mut tz = unsafe { mem::zeroed() };
+    let ret = unsafe { GetDynamicTimeZoneInformation(&mut tz) };
+    dbg!(ret);
+    match ret {
+        TIME_ZONE_ID_INVALID => {
+            return Err(crate::GetTimezoneError::IoError(
+                std::io::Error::last_os_error(),
+            ));
+        }
+        // XXX: I think this is
+        TIME_ZONE_ID_UNKNOWN => todo!(),
+        TIME_ZONE_ID_STANDARD | TIME_ZONE_ID_DAYLIGHT => {}
+        x => {
+            return Err(crate::GetTimezoneError::IoError(std::io::Error::other(
+                format!("Unexpected GetDynamicTimeZoneInformation() return `{x}`"),
+            )));
+        }
+    };
+    let windows_tz_id = &tz.TimeZoneKeyName;
+    let terminator = windows_tz_id.iter().position(|&c| c == 0).unwrap();
+    let windows_tz_id_str =
+        String::from_utf16(&windows_tz_id[..terminator]).map_err(std::io::Error::other)?;
+    dbg!(&windows_tz_id_str);
+
+    type UChar = u16;
+    #[repr(transparent)]
+    struct UErrorCode(i32);
+
+    // TODO: Link against windowsapp? Use rust_icu crate?
```

Comment:
I believe this can be provided by `windows-bindgen` since [it is defined in windows-sys](https://docs.rs/windows-sys/latest/windows_sys/Win32/Globalization/fn.ucal_getTimeZoneIDForWindowsID.html).

We'd want to link against the Windows-provided DLL here and not `rust_icu` since we want to take advantage of the automatically-updated `tzdb` Windows maintains in the background.